### PR TITLE
refactor user handling

### PR DESCRIPTION
**Pull request by franzbecker**

### CODE REVIEW

File: org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/DocumentProvider.xtend [hunk 1]
```diff
@@ -65,27 +71,29 @@ class DocumentProvider {
...
 		}
 	}
 
 	private def boolean create(File file) {
 		val parent = new File(file.parent)
 		if (!parent.exists) {
+			logger.debug("Creating directory='{}'.", parent)
 			parent.mkdirs
 		}
+		logger.debug("Creating file='{}'.", file)
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/DocumentProvider.xtend:102]: logged message is misleading, if called from createOrUpdate (line 43) on an existing file.

File: org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/workspace/WorkspaceResource.xtend [hunk 2]
```diff
@@ -23,78 +24,56 @@ import org.eclipse.jgit.api.errors.InvalidRemoteException
...
-		projectUrl = configuration.projectRepoUrl
-		separateUserWorkspaces = configuration.separateUserWorkspaces
-	}
+	@Inject Provider<User> userProvider
+	@Inject PersistenceConfiguration config
 
 	@GET
 	@Produces(MediaType.APPLICATION_JSON)
 	@javax.ws.rs.Path("list-files")
 	def Response listFiles(@Context HttpHeaders headers) {
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/workspace/WorkspaceResource.xtend]: headers still needed?

File: org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/workspace/WorkspaceResource.xtend [hunk 3]
```diff
@@ -111,16 +90,13 @@ class WorkspaceResource {
 	@Timed
 	@javax.ws.rs.Path("initialize")
 	def Response createWorkspace(@Context HttpHeaders headers) {
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/workspace/WorkspaceResource.xtend]: headers still needed?

File: org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/AbstractPersistenceTest.xtend [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+abstract class AbstractPersistenceTest {
+
+	@Inject Injector injector
+
+	@Inject protected extension AssertionHelper
+
+	@Before
+	def void performInjection() {
+		MockitoAnnotations.initMocks(this)
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/AbstractPersistenceTest.xtend:25]: I don't see mock annotations?

File: org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/AbstractPersistenceTest.xtend [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import com.google.inject.Module
+import com.google.inject.util.Modules
+import de.xtendutils.junit.AssertionHelper
+import java.util.List
+import javax.inject.Inject
+import org.junit.Before
+import org.mockito.MockitoAnnotations
+import org.testeditor.web.dropwizard.auth.User
+
+abstract class AbstractPersistenceTest {
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/AbstractPersistenceTest.xtend]: class comment that explains that injection module is prepared here to allow for user injection

File: org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/workspace/WorkspaceProviderTest.xtend [hunk 6]
```diff
@@ -1,46 +1,40 @@
...
 	@Test
 	def void getWorkspaceRoot() {
 		// given
-		val workspaceProvider = setupWorkspaceProvider(USE_SINGLE_WORKSPACES)
-		
+		config.separateUserWorkspaces = false
+
 		// when
-		val workspace = workspaceProvider.getWorkspace("theUser")
+		val workspace = workspaceProvider.getWorkspace()
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/workspace/WorkspaceProviderTest.xtend]: val workspace = workspaceProvider.workspace

File: org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/AbstractPersistenceTest.xtend (hunk 4 above)
Comment (franzbecker) [reply org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/AbstractPersistenceTest.xtend:25]: There are non yet - but could be used.

File: org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/DocumentProvider.xtend (hunk 1 above)
Comment (franzbecker) [reply org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/DocumentProvider.xtend:102]: The method is no longer called in the update case.

**Pull request merged**